### PR TITLE
Unit Test Documentation

### PR DESCRIPTION
**Pull request by paulhazen**
This PR introduces documentation regarding unit tests for the project.

#EOS-1903

### CODE REVIEW

File: com.playeveryware.eos/Documentation~/unit-testing.md
Comment (WispyMouse) [inline com.playeveryware.eos/Documentation~/unit-testing.md:1]: The name for this file is `unit-testing.md`. Should this be `unit_testing.md` to match other file names? (also update links to this if changed)

File: com.playeveryware.eos/Documentation~/unit-testing.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+## Scope of Unit Testing
+
+Unit tests should be written for **scripts used in the sample scenes** and for the **core plugin code**. The following guidelines outline the scope:
+
+### What Should Be Tested:
+- Core plugin functionality (e.g., helper classes, business logic, utilities, etc.).
+- Scripts attached to game objects or components in the sample scenes (but not the scenes themselves).
+
+### What Should Not Be Tested:
+- The **EOS SDK** itself. The SDK’s behavior is outside the scope of this project’s unit tests.
```
Comment (WispyMouse) [inline com.playeveryware.eos/Documentation~/unit-testing.md]: We do have some tests for Interfaces. Is that considered something we don't want to be testing?

File: com.playeveryware.eos/Documentation~/unit-testing.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+2. **PascalCase**: Use PascalCase for test method names, capitalizing each word.
+   - Example: `Login_WithInvalidPassword_ThrowsException()`
+
+3. **Group Tests by Class or Method**: Create test classes named after the class being tested, with `Tests` appended.
+   - Example: `OrderProcessorTests`
+
+4. **Arrange-Act-Assert**: Use comments or whitespace to clearly separate the three stages of each test.
+   - Example:
+   ```csharp
```
Comment (WispyMouse) [inline com.playeveryware.eos/Documentation~/unit-testing.md]: I'm surprised this works; you didn't define a end triple backtick thing, yet the markdown knows to end it properly after the double line break... But for my sanity, can you put an ending triple backtick?

**Review by WispyMouse — APPROVED** [review]
Seems good. Some tiny suggetions. My main takeaway from this is to have a single assert section, rather than "arrange, act, assert, arrange, act, assert". I think some my existing tests will need to broken down, then.

- Create a file, upload it. Test only this.
- Do the above without asserting. Download it. Test only this.
- 
etc

File: com.playeveryware.eos/Documentation~/unit-testing.md
Comment (paulhazen) [reply com.playeveryware.eos/Documentation~/unit-testing.md:1]: Makes sense. I'll change it to that and resolve this conversation once I have done so.

File: com.playeveryware.eos/Documentation~/unit-testing.md (hunk 1 above)
Comment (paulhazen) [reply com.playeveryware.eos/Documentation~/unit-testing.md]: We should change this to indicate that unit testing of the EOS SDK C# wrapper is not a current _priority_. It's not that it should _not_ be tested, just that it is not the current priority. Once the language has been updated to reflect that, I will mark this conversation as resolved.

File: com.playeveryware.eos/Documentation~/contributions.md [hunk 3]
```diff
@@ -59,6 +59,10 @@ A successful build will place the correct binaries in the proper locations for U
 
 See [standards.md](/com.playeveryware.eos/Documentation~/standards.md).
 
+## Unit Testing
+
+See [our documentation on unit testing](/com.playeveryware.eos/Documentation~/unit-testing.md)
```
Comment (paulhazen) [inline com.playeveryware.eos/Documentation~/contributions.md]: Add punctuation mark at the end of this and the same sections when added to other documents.

**Review by andrew-hirata-playeveryware — APPROVED** [review]

**Pull request merged**